### PR TITLE
Add a parameter to the configuration to allow the mailto link to be switched on or off

### PR DESCRIPTION
**Pull request by matrikular**
Pull Request replaces #9490

#### Summary of Changes
This PR allows to set the mailto link for each contact individually.

#### Test Instructions
1. **Backend:** Create a contact. Give it a title and make sure to fill out the email address field. The contact doesn't have to be assigned to a user or a category. For easy access, create a menu item to that contact in a menu of your choice.
2. **Frontend:** Verify that the email address including the mailto link is visible in the front-end. If not, please check the settings for com_contact.
3. **Backend:** Apply the patch, go to either the component, the contact or menu settings and set "Add Mailto: Link" to "No".
4. While the email address still is obfuscated, there should be no mailto: link.

### Documentation Changes Required
**Note: language files were changed and help screens might need to be updated.**

### CODE REVIEW

**Comment by C-Lodder** [comment]
Would personally this were done in a JHtml override rather than adding more parameters to Joomla. But just personal preference.

Has there been much demand for this?

File: administrator/components/com_contact/models/forms/contact.xml [hunk 1]
```diff
@@ -531,6 +531,18 @@
...
 
 			<field
+				name="add_mailto_link"
+				type="list"
+				label="COM_CONTACT_FIELD_PARAMS_ADD_MAILTO_LINK_LABEL"
+				description="COM_CONTACT_FIELD_PARAMS_ADD_MAILTO_LINK_DESC"
+				class="chzn-color"
+				useglobal="true"
+				>
+					<option value="1">JYES</option>
```
Comment (wojsmol) [inline administrator/components/com_contact/models/forms/contact.xml]: wrong indent

File: components/com_contact/views/contact/tmpl/default.xml [hunk 2]
```diff
@@ -119,6 +119,17 @@
 				<option value="1">JSHOW</option>
 			</field>
 
+			<field
+				name="add_mailto_link"
+				type="list"
+				label="COM_CONTACT_FIELD_PARAMS_ADD_MAILTO_LINK_LABEL"
+				description="COM_CONTACT_FIELD_PARAMS_ADD_MAILTO_LINK_DESC"
+				useglobal="true"
+			>
```
Comment (wojsmol) [inline components/com_contact/views/contact/tmpl/default.xml]: wrong indent

Comment (matrikular) [inline components/com_contact/views/contact/tmpl/default.xml]: @wojsmol can you elaborate on this please?

Comment (C-Lodder) [inline components/com_contact/views/contact/tmpl/default.xml]: The `>` needs another tab before

Comment (matrikular) [inline components/com_contact/views/contact/tmpl/default.xml]: The majority of the fields below / above is alligned this way.

Comment (C-Lodder) [inline components/com_contact/views/contact/tmpl/default.xml]: there is a PR that addresses this, so please use the same indentation

**Comment by matrikular** [comment]
@C-Lodder you're right. A developer could create an override for the JHtmlEmail::cloak method. Nevertheless, limiting the functionality of that override to a specific context can get messy quick.

An example of where this PR could be useful for non-developers as follows:
Say that you have two departments: Public Relations and a Customer Support. Both with their own representatives and individual contact forms (possible via plugin and / or custom fields). In case of the Customer Support, it is important to get the user to provide the required information, such as their customer number. Damage assessment could be another example.

One would want to guide the customer or visitor on the way he or she contacts you. That not only allows for a better, maybe even automatic processing of the request, but can also save every party a lot of time, preventing long dialogs about some missing information.

A / The mailto: link would bypass that guided path and while it is true that you could just decide not to show the email address on that very contact page, it seems a good thing to have the choice.

**Comment by franz-wohlkoenig** [comment]
@matrikular where to find:

> 3. Back-end: Go to the component settings and set "Add Mailto: Link" to "No"

At #9490 you wrote, ["Add Mailto: Link" is shown after apply Patch](https://github.com/joomla/joomla-cms/pull/9490#issuecomment-198877192). In this Issue (15973) you wrote to apply Patch on Step 5.

**Comment by franz-wohlkoenig** [comment]
@matrikular if this PR is ready for test, please giv me a hint.

**Comment by matrikular** [comment]
@franz-wohlkoenig thank you for your interest and pointing out the mistake in the description. As far as I'm concerned this is ready to test.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on e234d04baae979fdfb88d49b34001deb4370d9db<br /><br />Works as expected on Component, Contact and Menu.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15973">issues.joomla.org/tracker/joomla-cms/15973</a>.</sub>

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on e234d04baae979fdfb88d49b34001deb4370d9db<br /><br />I've tested this item successfully.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15973">issues.joomla.org/tracker/joomla-cms/15973</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

File: administrator/language/en-GB/en-GB.com_contact.ini [hunk 3]
```diff
@@ -169,6 +169,8 @@ COM_CONTACT_FIELD_NAME_DESC="Contact name."
 COM_CONTACT_FIELD_NAME_LABEL="Name"
 COM_CONTACT_FIELD_NUM_CONTACTS_DESC="Number of Contacts to display as list."
 COM_CONTACT_FIELD_NUM_CONTACTS_LABEL="Number of Contacts"
+COM_CONTACT_FIELD_PARAMS_ADD_MAILTO_LINK_LABEL="Add Mailto: Link"
+COM_CONTACT_FIELD_PARAMS_ADD_MAILTO_LINK_DESC="Adds a mailto: link to the displayed email address."
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_contact.ini]: Swap lines so they are in alphabetical order.

**Comment by infograf768** [comment]
@matrikular 
Bug (imho):
If I display a single contact AND set "Show Contact List" to display, then mailto is always present for each contact, whatever the settings for `add_mailto_link`

**Comment by matrikular** [comment]
@infograf768 thank you for reporting. I will look into it asap.

**Comment by matrikular** [comment]
@infograf768 it seems that there is an error in the view, where the link / route to each contact in that select list is build with a double encoded ampersand. The mailto link is only one of the "visible" symptoms. If you cycle through your contacts in the select list, you will notice that the breadcrumb navigation loses its "last" element as well.

If you may: #16268

Apart from that, I've fixed a mistake I made resovling a conflict in #16264.

**Pull request merged**